apps/extension/src/background/approvals/service.ts

```diff
@@ -34,7 +37,15 @@ export class ApprovalsService {
     protected readonly vaultService: VaultService,
     protected readonly chainService: ChainsService,
     protected readonly broadcaster: ExtensionBroadcaster
-  ) {}
+  ) {
+    browser.tabs.onRemoved.addListener((tabId) => {
+      const resolver = this.getResolver(tabId);
+      if (resolver) {
+        resolver.reject(new Error("Window closed"));
+        this.removeResolver(tabId);
```

Comment:
If the user closes the popup, it'll dispatch a message and clean the `resolverMap`

![Screenshot 2024-09-12 at 19 56 37](https://github.com/user-attachments/assets/454a1ffc-ce35-4c11-acaa-9a98468baa09)